ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/view/CircleImageViewTest.java

```diff
@@ -0,0 +1,379 @@
+import static org.powermock.api.support.membermodification.MemberModifier.suppress;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.ColorFilter;
+import android.graphics.Paint;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.ImageView.ScaleType;
+import com.kayako.sdk.android.k5.R;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({CircleImageView.class,ImageView.class,Paint.class,View.class})
+public class CircleImageViewTest {
+
+  private final static String INVALID_BOUNDS_MESSAGE = "adjustViewBounds not supported.";
+  private final int padLeft = 5;
+  private final int padTop = 5;
+  private final int padRight = 5;
+  private final int padBottom = 5;
+  private final int padStart = 5;
+  private final int padEnd = 5;
+  private final int width = 120;
+  private final int height = 240;
+  private static final int DEF_STYLE = 1;
+
+  private CircleImageView circleImageView;
+
+  @Mock
+  private Context context;
+
+  @Mock
+  private AttributeSet attributeSet;
+
+  @Mock
+  private Bitmap bitmap;
+
+  @Mock
+  private Drawable drawable;
+
+  @Mock
+  private Uri uri;
+
+  @Mock
+  private ColorFilter colorFilter;
+
+  @Rule
+  private ExpectedException thrown = ExpectedException.none();
+
+  @Before
+  public void setUp() throws Exception {
+    suppress(method(ImageView.class, "setScaleType"));
+    suppress(method(ImageView.class, "setImageBitmap"));
+    suppress(method(ImageView.class, "getDrawable"));
+    suppress(method(ImageView.class, "setImageDrawable"));
+    suppress(method(ImageView.class, "setImageResource"));
+    suppress(method(ImageView.class, "setImageURI"));
+    suppress(method(Paint.class, "setColor"));
+    suppress(method(Paint.class, "setColorFilter"));
+    suppress(method(View.class, "invalidate"));
+    suppress(method(View.class, "getWidth"));
+    suppress(method(View.class, "getHeight"));
+    suppress(method(View.class, "setPadding"));
+    suppress(method(View.class, "onSizeChanged"));
+    suppress(method(View.class, "setPaddingRelative"));
+
+    //circleImageView = new CircleImageView(context);
+    circleImageView = PowerMockito.spy(new CircleImageView(context));
+  }
+
+  @Test
+  public void borderColorTest() throws Exception {
+    // Arrange
+    int borderColor = circleImageView.getBorderColor();
+    borderColor++;
+
+    // Act
+    circleImageView.setBorderColor(borderColor);
+
+    // Assert
```

Comment:
✅ Good to see that the AAA("Arrange-Act-Assert") arranging and formatting pattern is followed.

Some of the main advantages are that it clearly separates what is being tested from the setup and verification steps. It also clarifies and focuses attention on a historically successful and generally necessary set of test steps.

**No change suggested, looks good!**	  

<a id="5927e2d0-4e70-11e8-8f04-8143e47b96ef" href="none" codereview-uuid-reserved-tag />